RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,213 @@
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+We'll need the `quant.sf` files for all the samples in an experiment which we have stored in `quant_dir`. 
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+```{r metadata-file}
+# sample metadata file
+meta_file <- file.path(data_dir, "gastric-cancer_metadata.tsv")
+```
+
+**Output**
+
+```{r output-names, live = TRUE}
+# Name the output gastric-cancer_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "gastric-cancer_tximeta.RDS")
+```
+
+## File names
+
+All output files from `salmon quant` we'll use with `tximeta` are named `quant.sf`.
+Unfortunately, this means that the file names themselves do not have any information about the sample they come from!
+
+```{r sf_files, live = TRUE}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` package.
+
+Notice how the file path is separated by `/`.
+If we were to split up this character string by `/`, the second to last item is the sample name.
+This is exactly what `stringr::word()` allows us to do: split up the file paths by `/` and extract the sample names.
+
+```{r sample_names}
+sample_names <- stringr::word(sf_files, -2, sep = "/")
+sample_names
+```
+
+## Set up metadata
+
+`tximeta` needs a data frame with at least these two columns: 
+- a `files` column  with the file paths to the quant.sf files
+- a `names` column with the sample names
+
+```{r names_sf_files}
+coldata <- data.frame(files = sf_files, 
+                      names = sample_names)
+```
+
+We have more information about these samples stored in the metadata file that we will also want stored in `coldata`. 
+Let's read in the sample metadata from the TSV file.
+
+```{r sample_meta_df, live = TRUE}
+# Read in the sample metadata TSV file and have a look
+sample_meta_df <- readr::read_tsv(meta_file)
+sample_meta_df
+```
+
+We'll want this information to be added to the `coldata`. 
+
+```{r join-sample_meta_df}
+coldata <- coldata %>%
+  dplyr::inner_join(sample_meta_df, by = c("names" = "srr_accession"))
+
+coldata
+```
+
+## Import expression data with `tximeta`
+
+Using the `coldata` data frame that we set up, we can now run the `tximeta()` to import our expression data while automatically finding and associating the transcript annotations that were used when we performed the quantification.
+
+The first time you run `tximeta()` you will get a message about storing the data in a [cache](https://en.wikipedia.org/wiki/Cache_(computing) so that it will retrieve the data faster the next time.
+We recommend you use the cache.
```

Comment:
I don't think the wikipedia link is very helpful, as it covers way more than a user needs to know here. 
```suggestion
The first time you run `tximeta()` you may get a message about storing downloaded transcriptome data in a cache directory so that it can retrieve the data more quickly the next time.
We recommend you use the cache, and accept the default location.
```